macosx/NSStringAdditions.mm

```diff
@@ -108,18 +108,23 @@ + (NSString*)stringForRatio:(CGFloat)ratio
 
 + (NSString*)percentString:(CGFloat)progress longDecimals:(BOOL)longDecimals
 {
+    NSNumberFormatter* formatter = [[NSNumberFormatter alloc] init];
+    formatter.numberStyle = NSNumberFormatterPercentStyle;
```

Comment:
1. As a method call that is frequent, we should make this formatter static.
2. To avoid threading issues between `maximumFractionDigits = 2` and `maximumFractionDigits = 1`, we should actually store two static formatters:
```
static NSNumberFormatter* longFormatter;
static NSNumberFormatter* shortFormatter;
static dispatch_once_t onceToken;
dispatch_once(&onceToken, ^{
    longFormatter = [[NSNumberFormatter alloc] init];
    longFormatter.numberStyle = NSNumberFormatterPercentStyle;
    longFormatter.maximumFractionDigits = 2;
    shortFormatter = [[NSNumberFormatter alloc] init];
    shortFormatter.numberStyle = NSNumberFormatterPercentStyle;
    shortFormatter.maximumFractionDigits = 1;
});
if (progress >= 1.0)
{
    return [shortFormatter stringFromNumber:@1)];
}
else if (longDecimals)
{
    return [longFormatter stringFromNumber:@(progress)];
}
else
{
    return [shortFormatter stringFromNumber:@(progress)];
}
```

Reply:
Applied your suggestions. Will push shortly. Thanks for the insight as well.

Reply:
Ah, this is neat. I made a version that applied `tr_truncd()` to `progress`, but apparently it's not necessary. Probably it was not correct either.

Reply:
Can you confirm that 100% is printed without fraction digits, since it looks like this version uses `maximumFractionDigits = 1` for it.

Reply:
Yep.

![Ekran Resmi 2023-06-06 03 31 03](https://github.com/transmission/transmission/assets/2769571/3130da59-2cb5-4b87-a88d-4a79fe3848d2)

Reply:
Great!